### DISCUSSION TITLE
Why's there no thread option?

### DISCUSSION THREAD

**Discussion by rew1nter** [projectdiscovery/cdncheck#444] [General]

**Comment by olearycrew** [comment]
@rew1nter can yuo describe in more detail what you're hoping to get out of a threading feature and how it would work?

**Comment by rew1nter** [comment]
There's DNS resolution going on under the hood. It's very slow for large data. Shouldn't we be able to set threads like dnsx?

**Comment by peter-kwan** [comment]
`subfinder` has the `-t` option to allow users to tweak the number of concurrent goroutines that it will use. On larger servers on AWS, I found that I can get 10x the speed with `-t 100`. So, I think this kind of flags is what @rew1nter was asking about.

I would find that useful too.

**Comment by ehsandeep** [comment]
@peter-kwan @rew1nter it's primarily used as library in [httpx](https://github.com/projectdiscovery/httpx), where you have `-t` and other controls to make it faster, but curious if it's diffrent use case to use CLI directly instead of using httpx?
